xapi-ingest-microservice/api.js

```diff
@@ -0,0 +1,245 @@
+            console.log({'err': err}, 'An error occured while getting the referenced learning activity statement');
+            return callback(err);
+          }
+
+          // Create the activity type summary (verb_objecttype)
+          var activityType = statement.verb.id.split('/').pop();
+          if (statement.object && statement.object.definition && statement.object.definition.type) {
+            activityType += '_' + statement.object.definition.type.split('/').pop();
+          } else if (refStatement && refStatement.object && refStatement.object.definition && refStatement.object.definition.type) {
+            activityType += '_' + refStatement.object.definition.type.split('/').pop();
+          }
+          activityType = activityType.toLowerCase();
+
+          // Get the user associated to this learning activity. If the user
+          // doesn't exist yet, it will be created
+          getOrCreateUser(ctx, statement, function(err, user) {
+            if (err) {
+              console.log({'err': err}, 'An error occured while getting the learning activity statement actor');
+              return callback(err);
+            }
+
+            // Store the learning activity statement in the DB
+            var storedStatement = {
+              'uuid': statement.id,
+              'statement': statement,
+              'verb': statement.verb.id,
+              'timestamp': statement.timestamp,
+              'activity_type': activityType,
+              'voided': false,
+              'tenant_id': ctx.auth.tenant_id,
+              'user_id': user.id,
+              'write_credential_id': ctx.auth.id
+            };
+
+            DB.Statement.create(storedStatement).complete(function(err, statement) {
+              if (err) {
+                console.log({'err': err}, 'Failed to store a new learning activity statement');
+                return callback({'code': 500, 'msg': err.message});
+              }
+
+              console.log({'statement': statement}, 'Sucessfully stored learning activity statement');
+              return callback();
+            });
+          });
+        });
+      });
+    })
+  }
+};
+
+/**
+ * Get a learning activity statement that's referenced in a different learning
+ * activity statement's object, if any
+ *
+ * @param  {Object}           statement                 The learning activity statement to check for referenced learning activities
+ * @param  {Function}         callback                  Standard callback function
+ * @param  {Object}           callback.err              An error that occurred, if any
+ * @param  {Statement}        callback.refStatement     The learning activity statement referenced in the provided learning activity statement
+ * @api private
+ */
+var getStatementRef = function(statement, callback) {
+  // The statement doesn't contain a statement reference
+  if (!statement.object || !(statement.object.objectType === 'StatementRef' && statement.object.id)) {
```

Comment:
The function checks if the a referenced learning activity is already available in the LRS. Hence, the combined condition to check if Object Type = StatementRef and there is an Id associated with it. The retrieval of the statement won't be possible without the Id. 

Also, the condition reappears in xapi-caliper utility and the main LRS application. Hence, keeping it consistent.